--- a/makefile
+++ b/makefile
@@ -17,8 +17,8 @@ COMPLDIR = $(shell pkg-config --variable
 MSGLANGS = $(notdir $(wildcard msg/*po))
 MSGOBJS = $(MSGLANGS:.po=/LC_MESSAGES/ncursesFM.mo)
 MSGFILES = $(LOCALEDIR)/$(MSGOBJS)
-LIBS =-lpthread -lmagic $(shell pkg-config --libs libarchive ncursesw libudev)
-CFLAGS =-D_GNU_SOURCE $(shell pkg-config --cflags libarchive ncursesw libudev) -DCONFDIR=\"$(CONFDIR)\" -DBINDIR=\"$(BINDIR)\" -DLOCALEDIR=\"$(LOCALEDIR)\"
+LIBS +=-lpthread -lmagic $(shell pkg-config --libs libarchive ncursesw libudev)
+CFLAGS +=-D_GNU_SOURCE $(shell pkg-config --cflags libarchive ncursesw libudev) -DCONFDIR=\"$(CONFDIR)\" -DBINDIR=\"$(BINDIR)\" -DLOCALEDIR=\"$(LOCALEDIR)\"
 
 # sanity checks for completion dir
 ifeq ("$(COMPLDIR)","")
@@ -97,10 +97,10 @@ objects-debug:
 	@cd $(SRCDIR); $(CC) -c *.c -Wall $(CFLAGS) -std=c99 -Werror -Wshadow -Wstrict-overflow -fno-strict-aliasing -Wformat -Wformat-security -g
 
 ncursesFM: objects
-	@cd $(SRCDIR); $(CC) -o ../ncursesFM *.o $(LIBS)
+	@cd $(SRCDIR); $(CC) -o ../ncursesFM *.o $(LDFLAGS) $(LIBS)
 
 ncursesFM-debug: objects-debug
-	@cd $(SRCDIR); $(CC) -o ../ncursesFM *.o $(LIBS)
+	@cd $(SRCDIR); $(CC) -o ../ncursesFM *.o $(LDFLAGS) $(LIBS)
 
 clean:
 	@cd $(SRCDIR); $(RM) *.o
