--- a/src/search.c
+++ b/src/search.c
@@ -45,7 +45,7 @@ void search(void) {
 
 static int recursive_search(const char *path, const struct stat *sb, int typeflag, struct FTW *ftwbuf) {
     char *fixed_str;
-    int len, r = 0, ret = FTW_CONTINUE;
+    int len, r = 0, ret = 0;
     /*
      * if searching "pippo" from "/home/pippo",
      * avoid saving "/home/pippo" as a result.
@@ -59,7 +59,7 @@ static int recursive_search(const char *
      * inside hidden folders
      */
     if (sv.search_lazy && fixed_str[0] == '.') {
-        return FTW_SKIP_SUBTREE;
+        return 2;
     }
     if ((sv.search_archive) && (is_ext(fixed_str, arch_ext, NUM(arch_ext)))) {
         return search_inside_archive(path);
@@ -74,10 +74,10 @@ static int recursive_search(const char *
         strncpy(sv.found_searched[sv.found_cont], path, PATH_MAX);
         sv.found_cont++;
         if (sv.found_cont == MAX_NUMBER_OF_FOUND) {
-            ret = FTW_STOP;
+            ret = 1;
         }
     }
-    return quit ? FTW_STOP : ret;
+    return quit ? 1 : ret;
 }
 
 /*
@@ -87,7 +87,7 @@ static int recursive_search(const char *
  */
 static int search_inside_archive(const char *path) {
     char *ptr;
-    int len = 0, ret = FTW_CONTINUE, r = 0, string_length;
+    int len = 0, ret = 0, r = 0, string_length;
     struct archive_entry *entry;
     struct archive *a = archive_read_new();
 
@@ -95,7 +95,7 @@ static int search_inside_archive(const c
     archive_read_support_format_all(a);
     string_length = strlen(sv.searched_string);
     if ((a) && (archive_read_open_filename(a, path, BUFF_SIZE) == ARCHIVE_OK)) {
-        while ((!quit) && (ret == FTW_CONTINUE) && (archive_read_next_header(a, &entry) == ARCHIVE_OK)) {
+        while ((!quit) && (ret == 0) && (archive_read_next_header(a, &entry) == ARCHIVE_OK)) {
             if (!sv.search_lazy) {
                 r = !strncmp(archive_entry_pathname(entry) + len, sv.searched_string, string_length);
             } else if (strcasestr(archive_entry_pathname(entry) + len, sv.searched_string)) {
@@ -105,7 +105,7 @@ static int search_inside_archive(const c
                 snprintf(sv.found_searched[sv.found_cont], PATH_MAX, "%s/%s", path, archive_entry_pathname(entry));
                 sv.found_cont++;
                 if (sv.found_cont == MAX_NUMBER_OF_FOUND) {
-                    ret = FTW_STOP;
+                    ret = 1;
                 }
             }
             ptr = strrchr(archive_entry_pathname(entry), '/');
@@ -115,12 +115,12 @@ static int search_inside_archive(const c
         }
     }
     archive_read_free(a);
-    return quit ? FTW_STOP : ret;
+    return quit ? 1 : ret;
 }
 
 static void *search_thread(void *x) {
     INFO("starting recursive search...");
-    nftw(ps[active].my_cwd, recursive_search, 64, FTW_MOUNT | FTW_PHYS | FTW_ACTIONRETVAL);
+    nftw(ps[active].my_cwd, recursive_search, 64, FTW_MOUNT | FTW_PHYS );
     if (!quit) {
         INFO("ended recursive search");
         if ((sv.found_cont == MAX_NUMBER_OF_FOUND) || (sv.found_cont == 0)) {
